### PR TITLE
ref bug #132451 Error email message, #130917 Message edit Group

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Fix validate email message
https://dev.framgia.com/redmine/issues/132451
Fix edit group message
https://dev.framgia.com/redmine/issues/130917

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java [hunk 1]
```diff
@@ -420,6 +420,10 @@ public void onPickGroup(View view) {
         if (mUser.getGroups() == null || mUser.getGroups().size() == 0) {
             return;
         }
+        if (!isVisibleGroup()) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: ERROR: [PMD-CPD] 21 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 2]
```diff
@@ -1242,6 +1242,10 @@ public void onPickGroup(View view) {
         if (mUser.getGroups() == null || mUser.getGroups().size() == 0) {
             return;
         }
+        if (!isEditGroupEnable()) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: [PMD-CPD] 21 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java

File: app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java (hunk 1 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: ERROR: [PMD-CPD] 19 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/requestovertime/RequestOvertimeViewModel.java

File: app/src/main/java/com/framgia/wsm/screen/requestovertime/RequestOvertimeViewModel.java [hunk 3]
```diff
@@ -285,6 +285,10 @@ public void onPickGroup(View view) {
         if (mUser.getGroups() == null || mUser.getGroups().size() == 0) {
             return;
         }
+        if (!isEditGroupEnable()) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestovertime/RequestOvertimeViewModel.java]: ERROR: [PMD-CPD] 19 lines duplicated.
app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -52,6 +52,7 @@
     <string name="invalid_email_format">Invalid email format !</string>
     <string name="least_6_characters">Please enter at least 6 characters !</string>
     <string name="confirm_password_does_not_match">Confirm password does not match !</string>
+    <string name="can_not_edit_group">Can not change Group</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`can_not_edit_group`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: pls add ``!`` character in end of this string

File: app/src/main/res/values-vi/strings.xml [hunk 5]
```diff
@@ -53,6 +53,7 @@
     <string name="least_6_characters">Vui lòng nhập ít nhất 6 ký tự !</string>
     <string name="confirm_password_does_not_match">Xác nhận mật khẩu không khớp !</string>
     <string name="request_leave">Yêu cầu đi trễ, về sớm</string>
+    <string name="can_not_edit_group">Không thể thay đổi phòng ban</string>
```
Comment (daolq3012) [inline app/src/main/res/values-vi/strings.xml]: pls add ``!`` character in end of this string

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -48,10 +48,11 @@
     <string name="confirm_password">Confirm password</string>
     <string name="forgot_password">Forgot Password?</string>
     <string name="log_in">Login</string>
-    <string name="is_empty">Empty, please enter !</string>
-    <string name="invalid_email_format">Invalid email format !</string>
-    <string name="least_6_characters">Please enter at least 6 characters !</string>
-    <string name="confirm_password_does_not_match">Confirm password does not match !</string>
+    <string name="is_empty">Empty, please enter!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`is_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -48,10 +48,11 @@
...
+    <string name="invalid_email_format">Invalid email format!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`invalid_email_format`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -48,10 +48,11 @@
...
+    <string name="least_6_characters">Please enter at least 6 characters!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`least_6_characters`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -48,10 +48,11 @@
...
+    <string name="confirm_password_does_not_match">Confirm password does not match!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`confirm_password_does_not_match`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -48,10 +48,11 @@
...
+    <string name="can_not_edit_group">Can not change Group!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`can_not_edit_group`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

**Pull request merged**